test/e2e/e2e_test.go

```diff
@@ -690,6 +683,88 @@ func checkMetricsCollection(t *testing.T, promClient prometheusv1.API, namespace
 	}
 }
 
+func runPromQueryForVector(t *testing.T, promClient prometheusv1.API, query string, sampleTime time.Time) (model.Vector, error) {
+	results, warnings, err := promClient.Query(context.Background(), query, sampleTime)
+	if err != nil {
+		t.Errorf("failed to execute prometheus query: %v", err)
+		return model.Vector{}, err
+	}
+	if len(warnings) > 0 {
+		t.Logf("prometheus query emitted warnings: %v", warnings)
+	}
+
+	result, ok := results.(model.Vector)
+	if !ok {
+		t.Fatalf("expecting scalar type result, found: %v ", reflect.TypeOf(results))
```

Comment:
I think a good rule of thumb for test helpers is to either accept testing.T and use `t.Fatalf` to fail, or return an error.  Mixing the 2 error handling strategies is likely to be confusing for future maintainers.

Reply:
Handled in the way mentioned in my previous comment